### DISCUSSION TITLE
Module loaders - Prós e Contras

### DISCUSSION THREAD

**Discussion by felipefialho** [frontendbr/forum#53] [Perguntas e Respostas]
Para segmentar melhor a discussão da issue #17.

**Comment by vitorbritto** [comment]
Atualmente temos WebPack, JSPM, Browserify e o Roll-up (que é utilizado no System.js como module bundler).

No meu caso, deixei o Bower de lado para trabalhar com o JSPM. Abri mão também do Browserify,  apesar de gostar muito do padrão CommonJS e o qual está presente também no NodeJS, pois estou trabalhando com o Babel (transpiler) que me permite codar e utilizar as facilidades do _modules_ do ES6 / Es.next / ES2015.

**Comment by renanvaz** [comment]
Vale citar que o JSPM engloba o System.js, que engloba o Rollup.

O JSPM além de module loader é tb um package manager.

**Comment by mksanti** [comment]
Pelo que vejo a galera esta utilizando mais WebPack e JSPM. 
Quem utiliza ou deixou de utilizar um para utilizar o outro, qual foi o motivo ?

**Comment by lagden** [comment]
Não conhecia o [Rollup](http://rollupjs.org/) e achei muito interessante!!

Não sou fã de utilizar transpilers (Babel, CoffeeScript, TypeScript, etc...), mas gostei do resultado do `bundle` gerado pelo **Rollup**.

No backend (**Node.js**) é possível utilizar bastante coisa de ES6 sem a necessidade de transpilar o código, já no front a história é um pouco diferente!! Bem que os navegadores estão acima 80% de compatibilidade, somente o Safari que ficou nos 50%.

Trabalho com [RequireJS](http://requirejs.org/) que já tem uma ferramenta de otimização `r.js`.
E quando rola um SPA é: `r.js` + [Almond](https://github.com/jrburke/almond).
